run_querytns.py

```diff
@@ -0,0 +1,225 @@
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+console = logging.StreamHandler()
+console.setFormatter(formatter)
+logger = logging.getLogger(__name__)
+if not logger.hasHandlers():
+    logger.addHandler(console)
+logger.setLevel(logging_level)
+
+# API key for Bot
+config = load_config()
+api_key = config.get('TNS', 'api_key')
+# PARAMETERS
+days_before_today = args.days_begin  # end = now - dbt
+days_to_include = args.days_end  # begin = now - dti
+z_min = args.zmin
+z_max = args.zmax
+months = args.limit_months  # pre-limit TNS search to candidates reported in the last X months
+if np.floor(days_before_today/30) > months:
+    logger.warning('Months limit restricts days_begin, consider increasing limit_months.')
+objtype = args.objtype  # Relevant TNS SN Ia subtypes.
+
+#--------------------------------------------------------------------
+# TNS API FUNCTIONS
+# Pre-provided helper functions for the TNS API
+# Obtained from https://wis-tns.weizmann.ac.il/content/tns-getting-started
+# function for changing data to json format
+url_tns_api = "https://wis-tns.weizmann.ac.il/api/get"
+def format_to_json(source):
+    # change data to json format and return
+    parsed = json.loads(source,object_pairs_hook=OrderedDict)
+    result = json.dumps(parsed,indent=4)
+    return result
+
+# function for search obj
+def search(url,json_list):
+    try:
+        # url for search obj
+        search_url = url+'/search'
+        # change json_list to json format
+        json_file = OrderedDict(json_list)
+        # construct the list of (key,value) pairs
+        search_data = [('api_key',(None, api_key)),
+                     ('data',(None,json.dumps(json_file)))]
+        # search obj using request module
+        response = requests.post(search_url, files=search_data)
+        # return response
+        return response
+    except Exception as e:
+        return [None,'Error message : \n'+str(e)]
+
+# function for get obj
+def get(url,json_list):
+    try:
+        # url for get obj
+        get_url = url+'/object'
+        # change json_list to json format
+        json_file = OrderedDict(json_list)
+        # construct the list of (key,value) pairs
+        get_data = [('api_key',(None, api_key)),
+                     ('data',(None,json.dumps(json_file)))]
+        # get obj using request module
+        response = requests.post(get_url, files=get_data)
+        # return response
+        return response
+    except Exception as e:
+        return [None,'Error message : \n'+str(e)]
+#--------------------------------------------------------------------
+#CALCULATE CURRENT DATE AND DATE RANGE TO SEARCH
+dt_end = datetime.date(datetime.now()) - timedelta(days=days_before_today)
+dt_begin = datetime.date(datetime.now()) - timedelta(days=days_to_include)
+date_begin = dt_begin.isoformat()
+date_end = dt_end.isoformat()
+
+#QUERY STRING
+query = "https://wis-tns.weizmann.ac.il/search?&discovered_period_value={}\
```

Comment:
Make the much more readable by splitting into `url` and `data`, where data is a `dict` of the keywords and parameters. These can then be passed directly to `requests.post(url, data=data)`. Requests will then make sure things are formatted correctly.

Reply:
This is just me being lazy because TNS does not provide this information. Also I wasn't sure if the keyword order somehow mattered. Let's leave this as an improvement: It seems redundant for now because 95% of these keywords are not used by us, but seem to appear when you search. I'm not even sure if the query will work without them provided as null?

Reply:
#15